src/main/java/filters/ExcludeTimeFilter.java

```diff
@@ -0,0 +1,85 @@
+package filters;
+
+import entities.Schedule;
+import entities.Section;
+import entities.Timeslot;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class filter checks whether a course overlaps with a specific time. The schedule does not pass the filter if
+ * it overlaps.
+ */
+public class ExcludeTimeFilter implements Filter {
+
+    private LocalTime lowerBound;
+    private LocalTime upperBound;
+    private Day filteredDay;
+
+    /**
+     * {@code filteredDay} defaults to Day.ALL_DAYS, meaning restriction to time range holds for all
+     * days.
+     *
+     * @see ExcludeTimeFilter(LocalTime, LocalTime, Day)
+     */
+    public ExcludeTimeFilter(LocalTime lowerBound, LocalTime upperBound) {
+        this(lowerBound, upperBound, Day.ALL_DAYS);
+    }
+
+    /**
+     * @param lowerBound exclusive Lower end of time range courses should be excluded from
+     * @param upperBound exclusive Upper bound of time range courses should be excluded from
+     * @param filteredDay Day to which these bounds are applied (Or all of them)
+     */
+    public ExcludeTimeFilter(LocalTime lowerBound, LocalTime upperBound, Day filteredDay) {
+        this.lowerBound = lowerBound;
+        this.upperBound = upperBound;
+        this.filteredDay = filteredDay;
+    }
+
+    /**
+     * @param s Schedule to check
+     * @return Whether some section conflicts with the time range
+     */
+    @Override
+    public boolean checkSchedule(Schedule s) {
+        if (s == null) {
+            return false;
+        }
+        List<Timeslot> timeslots = new ArrayList();
+
+        for (Section lec : s.getLectures()) {
```

Comment:
tbh this is repeated code and occurs in multiple filters. siddarth's pr adds a method to get all these slots, so when he merges it we should add that in everywehre